presqt/api_v1/serializers/resource.py

```diff
@@ -36,14 +77,70 @@ class ResourceSerializer(serializers.Serializer):
     size = serializers.IntegerField()
     hashes = serializers.DictField()
     extra = serializers.DictField()
-    download_url = serializers.SerializerMethodField()
+    links = serializers.SerializerMethodField()
+
+    def get_links(self, instance):
+        """
+        Translate the `links` property to a custom array of Hyperlink values.
+
+        Parameters
+        ----------
+        instance : Target Obj instance
+
+        Returns
+        -------
+        Hyperlink url for Target detail API endpoint
+        """
+        links = []
 
-    def get_download_url(self, instance):
-        reversed_url = reverse(
-            viewname='resource',
-            kwargs={
-                'target_name': self.context.get('target_name'),
-                'resource_id': instance['id'],
-                'resource_format': 'zip'})
+        resource_detail, resource_download, resource_upload = target_checker(
+            self.context.get('target_name'))
+
+        if resource_download is True:
+            reverse_download = reverse(
+                viewname='resource',
+                kwargs={
+                    'target_name': self.context.get('target_name'),
+                    'resource_id': instance['id'],
+                    'resource_format': 'zip'})
+            links.append({
+                "name": "Download",
+                "link": self.context['request'].build_absolute_uri(reverse_download),
+                "method": "GET"})
+
+        if resource_upload is True:
+            if instance['kind'] == 'container':
+                reversed_upload = reverse(
+                    viewname='resource',
+                    kwargs={'target_name': self.context.get('target_name'),
+                            'resource_id': instance['id']})
+                links.append({
+                    "name": "Upload",
+                    "link": self.context['request'].build_absolute_uri(reversed_upload),
+                    "method": "POST"})
+
+        return links
+
+
+def target_checker(target_name):
+    """
+    Checks in on targets.json and determines what actions are available for the requesting target.
+
+    Parameters
+    ----------
+    target_name: str
+
+    Returns
+    -------
+    """
+    with open('presqt/targets.json') as target_data:
```

Comment:
This is fine but I have been using a utility that opens a file...you can adopt this if you want to remain consistent otherwise it doesn't matter at all.